### PR TITLE
fix: fix adhocpopovers tab animate.

### PR DESCRIPTION
**Pull request by pkdotson**
### SUMMARY
The tab component within the adhocFilterEditpopvers were showing an animated view within the tab when user clicks on new tabs. This pr removes the default animate to false

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
before
https://user-images.githubusercontent.com/17326228/117086414-4a6a8700-ad01-11eb-9a1c-6f29b20c2d9c.mov

after

https://user-images.githubusercontent.com/17326228/117086431-55bdb280-ad01-11eb-9307-077a3d9a0ca3.mov

### TEST PLAN
Go to chart and open the popovers to ensure they do not animate when new tab is clicked.

### ADDITIONAL INFORMATION

- [ x] Has associated issue: https://github.com/apache/superset/issues/14475
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by ktmud** [comment]
Could we disable the animation globally by default?  E.g. like this:

```diff
diff --git a/superset-frontend/src/components/Tabs/Tabs.tsx b/superset-frontend/src/components/Tabs/Tabs.tsx
index 90c3a64e6..f016b73f3 100644
--- a/superset-frontend/src/components/Tabs/Tabs.tsx
+++ b/superset-frontend/src/components/Tabs/Tabs.tsx
@@ -20,81 +20,92 @@ import React from 'react';
 import { css, styled } from '@superset-ui/core';
 import AntDTabs, { TabsProps as AntDTabsProps } from 'antd/lib/tabs';
 import Icon from 'src/components/Icon';
+import { useTheme } from '@emotion/react';
 
 export interface TabsProps extends AntDTabsProps {
   fullWidth?: boolean;
   allowOverflow?: boolean;
 }
 
-const notForwardedProps = ['fullWidth', 'allowOverflow'];
+const StyledTabs = ({
+  animated = false,
+  fullWidth = true,
+  allowOverflow = true,
+  ...props
+}: TabsProps) => {
+  const theme = useTheme();
+  return (
+    <AntDTabs
+      animated={animated}
+      {...props}
+      css={css`
+        overflow: ${allowOverflow ? 'visible' : 'hidden'};
+
+        .ant-tabs-content-holder {
+          overflow: ${allowOverflow ? 'visible' : 'auto'};
+        }
 
-const StyledTabs = styled(AntDTabs, {
-  shouldForwardProp: prop => !notForwardedProps.includes(String(prop)),
-})<TabsProps>`
-  overflow: ${({ allowOverflow }) => (allowOverflow ? 'visible' : 'hidden')};
+        .ant-tabs-tab {
+          flex: 1 1 auto;
 
-  .ant-tabs-content-holder {
-    overflow: ${({ allowOverflow }) => (allowOverflow ? 'visible' : 'auto')};
-  }
+          &.ant-tabs-tab-active .ant-tabs-tab-btn {
+            color: inherit;
+          }
 
-  .ant-tabs-tab {
-    flex: 1 1 auto;
+          &:hover {
+            .anchor-link-container {
+              cursor: pointer;
 
-    &.ant-tabs-tab-active .ant-tabs-tab-btn {
-      color: inherit;
-    }
+              .fa.fa-link {
+                visibility: visible;
+              }
+            }
+          }
 
-    &:hover {
-      .anchor-link-container {
-        cursor: pointer;
+          .short-link-trigger.btn {
+            padding: 0 ${theme.gridUnit}px;
 
-        .fa.fa-link {
-          visibility: visible;
+            & > .fa.fa-link {
+              top: 0;
+            }
+          }
         }
-      }
-    }
-
-    .short-link-trigger.btn {
-      padding: 0 ${({ theme }) => theme.gridUnit}px;
-
-      & > .fa.fa-link {
-        top: 0;
-      }
-    }
-  }
-
-  ${({ fullWidth }) =>
-    fullWidth &&
-    css`
-      .ant-tabs-nav-list {
-        width: 100%;
-      }
-
-      .ant-tabs-tab {
-        width: 0;
-      }
-    `};
 
-  .ant-tabs-tab-btn {
-    display: flex;
-    flex: 1 1 auto;
-    align-items: center;
-    justify-content: center;
-    font-size: ${({ theme }) => theme.typography.sizes.s}px;
-    text-align: center;
-    text-transform: uppercase;
-    user-select: none;
-
-    .required {
-      margin-left: ${({ theme }) => theme.gridUnit / 2}px;
-      color: ${({ theme }) => theme.colors.error.base};
-    }
-  }
+        ${fullWidth &&
+        css`
+          .ant-tabs-nav-list {
+            width: 100%;
+          }
+
+          .ant-tabs-tab {
+            width: 0;
+            margin-right: 0;
+          }
+        `};
+
+        .ant-tabs-tab-btn {
+          display: flex;
+          flex: 1 1 auto;
+          align-items: center;
+          justify-content: center;
+          font-size: ${theme.typography.sizes.s}px;
+          text-align: center;
+          text-transform: uppercase;
+          user-select: none;
+
+          .required {
+            margin-left: ${theme.gridUnit / 2}px;
+            color: ${theme.colors.error.base};
+          }
+        }
 
-  .ant-tabs-ink-bar {
-    background: ${({ theme }) => theme.colors.secondary.base};
-  }
-`;
+        .ant-tabs-ink-bar {
+          background: ${theme.colors.secondary.base};
+        }
+      `}
+    />
+  );
+};
 
 const StyledTabPane = styled(AntDTabs.TabPane)``;
 
@@ -102,11 +113,6 @@ const Tabs = Object.assign(StyledTabs, {
   TabPane: StyledTabPane,
 });
 
-Tabs.defaultProps = {
-  fullWidth: true,
-  animated: true,
-};
-
 const StyledEditableTabs = styled(StyledTabs)`
   .ant-tabs-content-holder {
     background: white;

```

Note this diff also fixes a bug where tabs are not expanded to full width:

### Before

Xnip2021-05-04_19-20-51


### After

Xnip2021-05-04_19-20-18

**Comment by junlincc** [comment]
@pkdotson Thanks for the quick fix, let get the check pass!

@ktmud we need a good case to make that change. and even so, probably not in this PR.

**Review by geido — APPROVED** [review]
LGTM!

File: superset-frontend/src/explore/components/controls/FilterControl/AdhocFilterEditPopover/index.jsx [hunk 1]
```diff
@@ -162,6 +162,7 @@ export default class AdhocFilterEditPopover extends React.Component {
           data-test="adhoc-filter-edit-tabs"
           style={{ minHeight: this.state.height, width: this.state.width }}
           allowOverflow
+          animated={false}
```
Comment (zhaoyongjie) [inline superset-frontend/src/explore/components/controls/FilterControl/AdhocFilterEditPopover/index.jsx:165]: animated default value is `false`, so `animated` could be deleted here.

File: superset-frontend/src/explore/components/controls/MetricControl/AdhocMetricEditPopover/index.jsx [hunk 2]
```diff
@@ -348,6 +348,7 @@ export default class AdhocMetricEditPopover extends React.PureComponent {
           className="adhoc-metric-edit-tabs"
           style={{ height: this.state.height, width: this.state.width }}
           onChange={this.onTabChange}
+          animated={false}
```
Comment (zhaoyongjie) [inline superset-frontend/src/explore/components/controls/MetricControl/AdhocMetricEditPopover/index.jsx:351]: same before nit

**Review by zhaoyongjie — APPROVED** [review]
LGTM! with minor suggestions.

**Comment by rusackas** [comment]
@pkdotson please rebase. 

Also for the thread convo, I agree with the disabling of this particular animation _by default_ is a fine idea. We can enable it for any use cases where it makes sense.

See also https://github.com/apache/superset/pull/14496, which does exactly that.

File: superset-frontend/spec/javascripts/dashboard/components/DashboardBuilder_spec.jsx [hunk 3]
```diff
@@ -128,10 +128,11 @@ describe('DashboardBuilder', () => {
     expect(parentSize.find(Tabs.TabPane)).toHaveLength(2);
   });
 
-  it('should set animated=true on Tabs for perf', () => {
+  it('should have default animated=true on Tabs for perf', () => {
     const wrapper = setup({ dashboardLayout: undoableDashboardLayoutWithTabs });
     const tabProps = wrapper.find(ParentSize).find(Tabs).props();
-    expect(tabProps.animated).toEqual({ inkBar: true, tabPane: false });
+    console.log('tabProps', tabProps)
```
Comment (rusackas) [inline superset-frontend/spec/javascripts/dashboard/components/DashboardBuilder_spec.jsx:134]: Probably want to remove this console log. There CI linter is failing on the PR - this is likely one of the issues that needs addressing.

**Comment by rusackas** [comment]
@pkdotson please rebase, I think this is otherwise good to go.

**Pull request merged**